packages/rs-drive-light-client/src/proof/from_proof.rs

```diff
@@ -0,0 +1,208 @@
+use std::fmt::Debug;
+
+use crate::Error;
+use dapi_grpc::platform::v0::{self as platform};
+use dpp::prelude::{Identifier, Identity};
+pub use drive::drive::verify::RootHash;
+use drive::drive::Drive;
+
+use super::verify::verify_tenderdash_proof;
+
+// #[cfg(feature = "mockall")]
+
+/// Create an object based on proof received from DAPI
+pub trait FromProof<Req, Resp> {
```

Comment:
These parameters made as generics imply that there are possibility to have more that one implementation of the trait for a type, but is it our case? If not, it's better to have them as associated types

Reply:
as I see in this snippet they both return their own response types

Reply:
I mean yes, you're right there will be more than one `FromProof` implementation for `Identity`, but for each of implementation a pair of `Req`:`Resp` is fixed, so only one of them is a generic parameter

Reply:
Agreed with fominok (I think)